src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -45,12 +46,15 @@ internal abstract class PullRequestUpdater : IPullRequestUpdater
     private readonly IPcsVmrForwardFlower _vmrForwardFlower;
     private readonly IPcsVmrBackFlower _vmrBackFlower;
     private readonly ITelemetryRecorder _telemetryRecorder;
+    private readonly TelemetryClient _telemetryClient;
     private readonly ILogger _logger;
 
     protected readonly IReminderManager<SubscriptionUpdateWorkItem> _pullRequestUpdateReminders;
     protected readonly IReminderManager<PullRequestCheck> _pullRequestCheckReminders;
     protected readonly IRedisCache<InProgressPullRequest> _pullRequestState;
 
+    private const string PullRequestUpdateFailedEventName = "PullRequestUpdateFailed";
```

Comment:
I don't think this belongs in this class. It should be encapsulated somewhere else as it's the implementation detail of the telemetry.

Why not create a method in `TelemetryRecorder` where we usually encapsulate telemetry reporting?